### ISSUE TITLE
[BUG] pytest-django fails to recognize '-p no:django' when run in parallel ('-n X')

### ISSUE THREAD

**Issue by dybi** [pytest-dev/pytest-django#644]
When run in single process:
```bash
py.test -p no:django 
============================================================= test session starts ==============================================================
platform linux -- Python 3.6.6, pytest-3.6.2, py-1.5.4, pluggy-0.6.0
rootdir: /home/dybowski/GOLEM/concent/signing_service, inifile: pytest.ini
plugins: xdist-1.22.2, forked-0.2, cov-2.5.1, asyncio-0.8.0, celery-4.2.0
collected 37 items                                                                                                                             

test_handle_connection.py ......                                                                                                         [ 16%]
test_run.py ......                                                                                                                       [ 32%]
test_signing_service_methods.py ..................                                                                                       [ 81%]
test_utils.py .......                                                                                                                    [100%]

========================================================== 37 passed in 3.28 seconds ===========================================================
```
it works fine, but when run in parallel:
```bash
py.test -p no:django -n 4
============================================================= test session starts ==============================================================
platform linux -- Python 3.6.6, pytest-3.6.2, py-1.5.4, pluggy-0.6.0
rootdir: /home/dybowski/GOLEM/concent/signing_service, inifile: pytest.ini
plugins: xdist-1.22.2, forked-0.2, cov-2.5.1, asyncio-0.8.0, celery-4.2.0
gw0 [37] / gw1 [37] / gw2 [37] / gw3 [37]
scheduling tests via LoadScheduling
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE                                                                                                    [100%]
==================================================================== ERRORS ====================================================================
 ERROR at setup of TestSigningServiceHandleConnection.test_that__handle_connection_should_send_golem_message_signed_transaction_if_frame_is_correct 
[gw0] linux -- Python 3.6.6 /home/dybowski/ENVs/concent/bin/python

request = <SubRequest 'django_test_environment' for <Function 'test_that__handle_connection_should_send_golem_message_signed_transaction_if_frame_is_correct'>>

    @pytest.fixture(autouse=True, scope='session')
    def django_test_environment(request):
        """
        Ensure that Django is loaded and has its testing environment setup.
    
        XXX It is a little dodgy that this is an autouse fixture.  Perhaps
            an email fixture should be requested in order to be able to
            use the Django email machinery just like you need to request a
            db fixture for access to the Django database, etc.  But
            without duplicating a lot more of Django's test support code
            we need to follow this model.
        """
        if django_settings_is_configured():
            _setup_django()
            from django.conf import settings as dj_settings
            from django.test.utils import (setup_test_environment,
                                           teardown_test_environment)
>           dj_settings.DEBUG = False

/home/dybowski/ENVs/concent/lib/python3.6/site-packages/pytest_django/plugin.py:353: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/dybowski/ENVs/concent/lib/python3.6/site-packages/django/conf/__init__.py:70: in __setattr__
    super(LazySettings, self).__setattr__(name, value)
/home/dybowski/ENVs/concent/lib/python3.6/site-packages/django/utils/functional.py:268: in __setattr__
    self._setup()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <LazySettings [Unevaluated]>, name = None

    def _setup(self, name=None):
        """
            Load the settings module pointed to by the environment variable. This
            is used the first time we need any settings at all, if the user has not
            previously configured the settings manually.
            """
        settings_module = os.environ.get(ENVIRONMENT_VARIABLE)
        if not settings_module:
            desc = ("setting %s" % name) if name else "settings"
            raise ImproperlyConfigured(
                "Requested %s, but settings are not configured. "
                "You must either define the environment variable %s "
                "or call settings.configure() before accessing settings."
>               % (desc, ENVIRONMENT_VARIABLE))
E           django.core.exceptions.ImproperlyConfigured: Requested settings, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.

/home/dybowski/ENVs/concent/lib/python3.6/site-packages/django/conf/__init__.py:39: ImproperlyConfigured
```
so it seems that `-p no:django` is not recognized correctly/taken into account when `py.test` runs in parallel mode.

Packages versions:
```bash
pip freeze| grep pytest

pytest==3.6.2
pytest-asyncio==0.8.0
pytest-cov==2.5.1
pytest-django==3.3.2
pytest-forked==0.2
pytest-xdist==1.22.2

```

**Comment by blueyed** [comment]
pytest-django is disabled according to

> plugins: xdist-1.22.2, forked-0.2, cov-2.5.1, asyncio-0.8.0, celery-4.2.0

But somehow it still gets invoked then?!

Looks more like a problem with pytest-xdist, which maybe does not forward/handle that arg correctly, and then the childs are using the plugin?

**Comment by dybi** [comment]
shall I file another ticket on `pytest-xdist`? Or how do we proceed from here?

**Comment by blueyed** [comment]
I would say so, yes - but maybe @RonnyPfannschmidt can say something here already?

**Comment by RonnyPfannschmidt** [comment]
this is a bug in both pytest and pytest-xdist - since a few years now -p no:... cannot pass to xdist at all ever due to a config initialization order issue

**Comment by dybi** [comment]
@RonnyPfannschmidt - **any** chances of having it fixed? Or it's a no-fix type of bug?

**Comment by RonnyPfannschmidt** [comment]
@dybi its the type of bug that needs between a full work week to a full work months to fix since its petty deep in the guts of pytest - getting this fixed is on the radar in https://github.com/pytest-dev/pytest/projects/2 but it will be a while before it fully materealizes

**Comment by dybi** [comment]
ok, so I will keep my fingers crossed ;) or is there anything I can do to help?

**Comment by blueyed** [comment]
Closing it here then.
Thanks for reporting and providing feedback!
